### PR TITLE
Check for working hours + rename isAvailable

### LINKED ISSUE #5331: [CAL-305] Round robin assigns to unavailable host
If a host is not available due to their working hour settings, round robin still includes it into the pool which can cause the event to be assigned to an unavailable round robin host.

> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by emrysal**
Fixes #5331

Consider this a hotfix, I'm going to attempt and split apart the schedule and busyTime logic going forward.

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -83,40 +84,56 @@ async function refreshCredentials(credentials: Array<Credential>): Promise<Array
...
+      timeSlotEnd.isBetween(startTime, endTime, null, "(]")
+    ) {
+      return true;
+    }
+  });
+  return false;
+};
 
+// if true, there are conflicts.
+function checkForConflicts(busyTimes: BufferedBusyTimes, time: dayjs.ConfigType, length: number) {
```
Comment (emrysal) [inline packages/features/bookings/lib/handleNewBooking.ts:114]: This clarifies its function over the current naming.

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 2]
```diff
@@ -83,40 +84,56 @@ async function refreshCredentials(credentials: Array<Credential>): Promise<Array
...
-    i++;
+  busyTimes.forEach((busyTime) => {
     const startTime = dayjs(busyTime.start);
     const endTime = dayjs(busyTime.end);
-
     // Check if time is between start and end times
     if (dayjs(time).isBetween(startTime, endTime, null, "[)")) {
-      t = false;
-      break;
+      return true;
```
Comment (emrysal) [inline packages/features/bookings/lib/handleNewBooking.ts:125]: Early return replaces result boolean

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 3]
```diff
@@ -83,40 +84,56 @@ async function refreshCredentials(credentials: Array<Credential>): Promise<Array
...
+  {
+    workingHours,
+  }: {
+    workingHours: WorkingHours[];
+  }
+) => {
+  const timeSlotStart = dayjs(timeSlot.start).utc();
+  const timeSlotEnd = dayjs(timeSlot.end).utc();
+  workingHours.forEach((workingHour) => {
+    // TODO: Double check & possibly fix timezone conversions.
```
Comment (emrysal) [inline packages/features/bookings/lib/handleNewBooking.ts:98]: In an ideal world the current logic already works, but..

**Review by alishaz-polymath — APPROVED** [review]
👍

**Pull request merged**